### ISSUE TITLE
v2.0.22 UWTROCK are very dominant especially when zoomed out.

### ISSUE THREAD

**Issue by Hakansv** [bdbcat/o-charts_pi#50]
On Win10 using the current master. 
o-charts_pi updated today to 2.0.22 (from 2.0.13)

The underwater rocks symbol have changed in this version. (To S52 standard?) Was this intentional and if so why?
What's more critical is the crowded not scaled view while zoomed out. See pict.
My Win is scaled to 175% but a similar view is also on a Win 100% scaled.

For info was the first view after plugin update showing enormous soundings font size. About 50 mm height! After some zooming,  lights toggle and more they gradually returned to normal.

o-charts v2.0.13 oesu (the same on pure ENC) to the left and o-charts v2.0.22, oesu to the right
![bild](https://github.com/user-attachments/assets/745426b4-752d-4f2e-bbb9-8507c81d8a78)

oesu zoomed out.
![bild](https://github.com/user-attachments/assets/a0024026-71e8-4dcd-9960-748fd3bd865d)

**Comment by bdbcat** [comment]
Do you have SCAMIN enabled?
Options->Charts->Vector Chart Display->Reduced Detail at small scale.

**Comment by Hakansv** [comment]
Yes I have.
But there are differences for the two canvas. Pict 1.


![bild](https://github.com/user-attachments/assets/eeb0c359-f2db-4e1b-95ad-90dc6aaed0ad)

**Comment by bdbcat** [comment]
I think I am missing your point.

**Comment by Hakansv** [comment]
Three pictures from smaller scale to bigger.
The in between has no symbols but the two other has. See the view scale on each

**Comment by bdbcat** [comment]
Hmmmm...

**Comment by Hakansv** [comment]
And why the decision to change symbol? I'm just curious.

**Comment by Hakansv** [comment]
Also other objects are show on smaller scale than before:
Like traffic separation zones. And BOYSPP has grown.

![bild](https://github.com/user-attachments/assets/178c864d-1ae8-4b97-b2cf-15050bcccfa9)

**Comment by bcn2** [comment]
Don't see that issue neither on our Mac nor on the W10 system, both with the latest 2.0.22 - however changing the "Display Category" at Chart Panel Options, this makes a difference. "All" and "User Standard" for example.
The TSS are ok in any case.

**Comment by bdbcat** [comment]
I seems as though maybe our view scale calculation is not quite correct, maybe latitude dependent.
To check this, create a 1.0 NMi vertical route segment.  Measure the physical length of this segment on YOUR screen, using a caliper, or so.  Do the calculation.   Display scale = 1852m / {length on screen, m}
 By my measurements, we seem to be somewhat in error.

Approved ECDIS systems specify the screen resolution (pixels per mm) pretty strictly.  We have to adjust to a much wider variety of displays.

This change arose on studying a particular WRECK in France.  We discovered an error in our S52 PLIB implementation.  Corrected that to pure S52 requirements, and this is what you see.

But again, I agree that results are not good in SE.  Lots of awash rocks there. 

I agree the some areas in SE have become overly dense with Isolated Danger symbols.

**Comment by bcn2** [comment]
![Captura de pantalla 2024-11-14 a las 21 36 47](https://github.com/user-attachments/assets/690e7a78-d2eb-42d8-8e06-e9f98225f401)

**Comment by Hakansv** [comment]
Measurement:
1852/0.0564 = 32836.88
View scale = 1:33000

![bild](https://github.com/user-attachments/assets/b637c113-b697-41f2-8105-74854c87db24)

**Comment by bdbcat** [comment]
Pretty close.
I wonder if, contrary to Mercator paper nav methods, we are meant to compare SCAMIN value to a horizontal scale?
Would make a big difference in SE.
I'll explore...

**Comment by Hakansv** [comment]
Here at about 57° N
Horizontal route:
1852/0.0504 = 36746.0
View scale = 1:33000

**Comment by bdbcat** [comment]
Taking a simplistic approach:
I have paper chart NOAA 11300.  A relic of the past. Chart scale (printed on the paper chart) is 1:460,000.
I take my trusty brass navigator's divider, measure 15 NMi on latitude scale of paper chart.
I sit at OPCN, zoom in to the same area as depicted on the paper chart.  Using the unchanged divider setting, I adjust OCPN zoom so that 15 miles vertically matches the divider setting.  I read the OCPN view scale,
Result: 709, 000.
Something definitely wrong here.

**Comment by Hakansv** [comment]
FYI: I updated to v2.0.23
First view after plugin update, first shot.
After a O restart, second shot.

![bild](https://github.com/user-attachments/assets/f1b2df75-f0b1-4011-8794-8d3f94376bbb)

![bild](https://github.com/user-attachments/assets/ca8f5510-a793-489e-86ac-ce5a46485c5e)

**Comment by bcn2** [comment]
.23 is addressing a different issue - problems with online log-in for new installs on Win

**Comment by bdbcat** [comment]
I found that the auto measure of screen size (mm) for my linux GTK system is not correct.  This of course distorted the calculation of screen scale.  Fixed.

Now I see that if the Settings->Advanced->zoom-scale/vector is set to 0, then the SCAMIN is followed as per the S52 spec.
However, if zoom-scale is > 0, then SCAMIN calculations are not correct, and this is a bug.  I'll fix that in o-charts_pi.

Now speaking about the danger symbol on o-charts_pi using the SE chartset specifically:
The archipelagos of Sweden contain a lot of "awash" rocks, and these have been faithfully captured in the ENCs.  I have investigated the SCAMIN values on these features, and find that the Swedish HO follows the S57 recommended standard.  That is, the SCAMIN value is roughly 3 or 4 times the original chart compilation scale.

There is a nice description from David Burch here:  http://davidburchnavigation.blogspot.com/2021/09/SCAMIN.html

Fiddling around a bit, I find that a more practical navigation display can be obtained by _reducing_ the zoom-scale value to -1, or even -2.  What this does in OCPN is this:  On zoom in, a negative zoom-scale _delays_ the switch to a larger scale chart slighly.  This has the effect of reducing clutter from danger symbol.

Also note:  The display category (base/standard/all) has no effect on the danger symbol.  S52 specs require that danger symbols be shown on all categories selected, as a safety measure.  It is not possible to filter them out, except by scamin.

Now, we could decide to deviate from S52 in the interest of practical OCPN usability incases such as this.  After all, we are mostly on small boats, mostly navigating in-shore using large-scale charts.  A slightly different use-case from larger vessels offshore or in well charted port regions.

Thoughts?

If it were possible, I sure would like to see the behavior of a type-certified ECDIS in similar situation.

**Comment by Hakansv** [comment]
Thanks for the link to the "Burch page". Very informative although, what I understand, the Table 2.2 on that page don't comply with the SE HO use where we can find a SCAMIN at 50000-1 for the UWTROCK (WATLEV 1-5). 

So..
The first step for us would be to actually use the stated SCAMIN to not show these objects when the display scale is >=50000 as in this UWTROCK case.
See my shot below. If I use no-quilt on that position the UWTROCK  are drawn on file OC-46-AL7IE5 only. The compilation scale is 1:12000 and SCAMIN 50000-1. And we do handle the WATLEV 1-3 and 4-5 with different symbols. (Although I don't like the danger symbol for UWTROCK-04/05 but that's only a personal view :)) 
If we zoom out here from display scale 1:12000 neither UWTROCK would be drawn at a display scale >50000-1, but they are. And here O are not fooled by the same object on other cells for the same position.

Second step: 
> we could decide to deviate from S52 in the interest of practical OCPN usability in cases such as this.

Yes, I do agree on that. 
One way is of course to go back to the "old" underwater UWTROCK 04/05 symbol but that's not feasible I suppose.

Another way would be to decrease the danger symbol size if the display scale is x times the compilation scale up to SCAMIN where it's hidden. Example: Symbol scale size times display scale / compilation scale.  (e.g: size scale * 0.6 for  1:20000/1:12000 in the object case for my shot below.) The multiplier must of course be tested for an appropriate value.

BTW:  `The archipelagos of Sweden contain a lot of "awash" rocks` 
True, but it may be Finland has more and there are a lot rocks in Norway as well.

As usual: pls let me try again if this is not understandable.


![bild](https://github.com/user-attachments/assets/a76b6038-42ed-425a-8947-af4aba5afa5e)

**Comment by Hakansv** [comment]
For info:
The visual difference for UWTROCK 1-3 and UWTROCK 4-5 is that WATLEV 1-3 are possible visible by braking waves or disturbed water movement but 4-5 are not visible and though more "danger".

**Comment by Hakansv** [comment]
For info 2:
I've asked a fellow for a ECDIS picture but not sure when it can happen.

**Comment by Hakansv** [comment]
And I suppose it's clear this discussion is about all SCAMIN?
Here BOYSPP with scamin 118000-1
Both oversized and for this display scale it should not have been visible.
![bild](https://github.com/user-attachments/assets/7d6bcbb5-2496-47ed-9c33-f71a9d37043c)

**Comment by RooieDirk** [comment]
Apart from de SCAMIN problem, wich I cant reproduce here with the latest master and O-23, I think there is the wrong symbol in use for the rocks. The rocks in sweden shown with the isolated danger symbol often have a depth 'unknown'. According chart-1 the ECDIS symbol should be different. (I think;-))
![Screenshot_20241120_124101](https://github.com/user-attachments/assets/77243b41-77a6-4209-a675-ae89eabd60da)

**Comment by Hakansv** [comment]
Yes, we saw that possible option as well. Open for future considerations but not needed for this issue:
Dave has now read and penetrated the standard in all its intricacies and the Isolated Danger symbol is not needed in these shallow waters. Just as you indicated.
"Case closed"
